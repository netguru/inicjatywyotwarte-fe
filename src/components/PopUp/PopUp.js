import React, { useState } from 'react'
import styled from 'styled-components'
import SummaryButton from 'components/Buttons/SummaryButton/SummaryButton'

const Container = styled.div`
  position: fixed;
  bottom: 10px;
  left: 0;
  right: 0;
  margin: 0 auto;
  max-width: 90%;
  padding: 25px;
  font-size: 0.9rem;
  background: ${props => props.theme.mainColorCategoryNeighbour};
  color: ${props => props.theme.fontColorMedium};
  z-index: 10;
  opacity: 0.97;
  text-align: justify;
  line-height: 1.4rem;
  border-radius: ${props => props.theme.borderRadiusMedium};

  ${props => !props.shouldShowPopUp && 'display: none;'}
`
Container.displayName = 'PopUpContainer';

const StyledSummaryButton = styled(SummaryButton)`
  margin-bottom: 0;
`

const PopUp = ({ content }) => {
  const [shouldShowPopUp, setShouldShowPopUp] = useState(true)

  const handlePopUpClose = () => {
    localStorage.setItem('areCookiesAccepted', 'true')
    setShouldShowPopUp(false)
  }

  return (
    <Container shouldShowPopUp={shouldShowPopUp}>
      {content}
      <StyledSummaryButton onClick={handlePopUpClose}>
        AkceptujÄ™
      </StyledSummaryButton>
    </Container>
  )
}

export default PopUp
