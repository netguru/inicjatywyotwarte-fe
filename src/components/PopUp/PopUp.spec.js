import React from 'react'
import PopUp from './PopUp'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('PopUp', () => {
  test('renders mocked icon with proper title', () => {
    const wrapper = shallow(<PopUp/>);
    expect(wrapper.find({ shouldShowPopUp: true }).length).toEqual(1);
    expect(wrapper.find('Styled(SummaryButton)').text()).toEqual('AkceptujÄ™');
  });

  test('shouldShowPopUp changes after clicking button', () => {
    const wrapper = shallow(<PopUp/>);
    wrapper.find('Styled(SummaryButton)').simulate('click');
    expect(wrapper.find({ shouldShowPopUp: false }).length).toEqual(1);
  });

  test('match snapshot', () => {
    const wrapper = shallow(<PopUp/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


