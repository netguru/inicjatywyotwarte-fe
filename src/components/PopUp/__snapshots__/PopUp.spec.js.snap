// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp match snapshot 1`] = `
<PopUpContainer
  shouldShowPopUp={true}
>
  <Styled(SummaryButton)
    onClick={[Function]}
  >
    AkceptujÄ™
  </Styled(SummaryButton)>
</PopUpContainer>
`;
